miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -66,6 +66,21 @@ proc.authenticate_account_origin
     # => [storage_offset, ...]
 end
 
+#! Checks that the invocation of a procedure originates from the native account.
+#! This procedure will prevent usage of the account procedures which can mutate the account state 
+#! with the foreign accounts.
+#!
+#! Panics:
+#!   - if the invocation of the account procedure does not originate from the native account.
+#!
+#! Stack: []
+#! Output: []
+proc.check_native_account
+    exec.memory::get_account_id
+    exec.memory::get_native_account_id
+    assert_eq
+end
```

Comment:
nit: I would probably rename this into `assert_native_account`.

Also, I think we could do this with a single memory read by checking if account data pointer is 2048 or not.